simulator/core/perf_sim.cpp

```diff
@@ -65,10 +68,10 @@ typename PerfSim<ISA>::FuncInstr PerfSim<ISA>::read_instr(Cycle cycle)
         return rp_decode_2_decode->read( cycle);
     }
     const auto& _data = rp_fetch_2_decode->read( cycle);
-    FuncInstr instr( _data.raw,
-        _data.PC,
-        _data.predicted_taken,
-        _data.predicted_target);
+    FuncInstr instr( _data.raw, 
```

Comment:
I think you may add a new ctor to FuncInstr for that.

Reply:
Pavel Igorevich, may I ask you? How is it possible to create MIPSInstr like
 MIPSInstr( fetch( PC), PC);
if our ctor is 
MIPSInstr::MIPSInstr( uint32 bytes, Addr PC,
                      bool predicted_taken,
                      Addr predicted_target) 
It is in mips_memory.cpp

Reply:
Add one more ctor.
```c++
MIPSInstr::MIPSInstr( uint32 bytes, const BPInterface& bp_info) :
      MIPSInstr( bytes, bp_info.branch_ip, bp_info.is_taken, bp_info.target) { }
```